allennlp/training/metrics.py

```diff
@@ -288,3 +290,196 @@ def reset(self):
         self._true_negatives = 0.0
         self._false_positives = 0.0
         self._false_negatives = 0.0
+
+
+@Metric.register("conll_span_f1")
```

Comment:
On second thought: another option is to just allow all metrics to take the `Vocabulary` in their `from_params` method.  If you do that, along with switching the ignored classes from ints to strings, it'd be relatively straightforward to construct this `from_params`.  That's probably the better way to go.